Improve autonomous database Micronaut guide and fix several issues (#1088)
micronaut-projects/micronaut-guides: guides/micronaut-oracle-autonomous-db/micronaut-oracle-autonomous-db.adoc

Create a test to verify that database access works:

test:repository/ThingRepositoryTest[]

== Testing the Application

There are two options for running the tests; one is to run against the live database, and the other is to run tests locally with an Oracle database in a Docker container using https://www.testcontainers.org/[Testcontainers].

> Replace the following dependencies:

Reviewer: minor - change "dependencies" to "dependency", and remove both `:dependencies:` - that's only needed if you have 2 or more dependencies that you want to render together

:dependencies:
dependency:junit-jupiter[groupId=org.testcontainers,scope=testImplementation]
:dependencies:

with:

> :dependencies:

Reviewer: and since there's only 1 dependency here you can remove the two `:dependencies:`

Author: Thanks

{"message":"Page Not Found","_links":{"self":{"href":"/things/Ralph","templated":false}}}
```

Run the application again, and everything should work with the password placeholders replaced with the resolved secret values at startup.

> == Connect the db with the IntelliJ Database tool

Reviewer: "db" -> "database"

Update your Autonomous Database Instance to Allow both TLS and mTLS Authentication:

In our Oracle Cloud Autonomous Database Details page under Network section click on "Access control list" "Edit" link.

image::oracleautodb/dbtool1.png[]

{empty} +

In opened dialog "Edit Access Control List" for "IP notation type" select box choose "IP Address", then click on "Add My IP Address" Button and "Save Changes".

image::oracleautodb/dbtool2.png[]

{empty} +

In one field below  "Mutual TLS (mTLS) authentication:" click on "Edit" link.
Uncheck "Require mutual TLS (mTLS) authentication" check box in opened "Edit Mutual TLS Authentication" dialog and "Save Changes".

image::oracleautodb/dbtool3.png[]

{empty} +

Click on "DB Connection' button then choose one of Connection Strings and click "Copy" link. It will be used for creating connection url.

image::oracleautodb/dbtool4.png[]

{empty} +

Setup DB connection in the IntelliJ Database tool:

Select Database tool and Data Source Properties in IntelliJ

image::oracleautodb/dbtool5.png[]

{empty} +

In opened Data Source and Drivers window select adding new DataSource and choose Oracle

image::oracleautodb/dbtool6.png[]

{empty} +

> In Configuration window add "Name"": MicronautDemo, "User": micronautdemo and "Password" specified before.
> In "URL" field remove "localhost:1521:XE" and add previously copied Connection String

Reviewer: ```suggestion
In the Configuration window, set *Name* to "MicronautDemo", *User* to "micronautdemo", and *Password* to the schema user password. Replace the *URL* value "localhost:1521:XE" with the previously copied Connection String.
```

image::oracleautodb/dbtool7.png[]

{empty} +

> URL field should looks like: jdbc:oracle:thin:@(description= (retry_count=20) .... L=Redwood City, ST=California, C=US")))

Reviewer: ```suggestion
The URL field should look like: `jdbc:oracle:thin:@(description= (retry_count=20) .... L=Redwood City, ST=California, C=US")))`
```

> Click on Test Connection and you should get "Secceeded" message, then click "Ok"

Reviewer: ```suggestion
Click *Test Connection* and you should see a "Succeeded" message, then click *OK*.
```

image::oracleautodb/dbtool8.png[]

{empty} +

> Under MICRONAUTDEMO user make select query of 'THING' table

Reviewer: ```suggestion
Under the `MICRONAUTDEMO` user, create a select query from the `THING` table:
```
